src/transformers/generation/flax_utils.py

```diff
@@ -626,21 +646,28 @@ def greedy_search_body_fn(state):
             return GreedyState(
                 cur_len=state.cur_len + 1,
                 sequences=next_sequences,
+                scores=tokens_scores,
                 running_token=next_token,
                 is_sent_finished=next_is_sent_finished,
                 model_kwargs=next_model_kwargs,
             )
 
         # The very first prompt often has sequence length > 1, so run outside of `lax.while_loop` to comply with TPU
-        if input_ids.shape[1] > 1:
+        # Besides, when output_scores is true, to return scores vocab_size of the model is got from first run.
+        if input_ids.shape[1] > 1 or output_scores:
             state = greedy_search_body_fn(state)
 
         if not trace:
             state = self._run_loop_in_debug(greedy_search_cond_fn, greedy_search_body_fn, state)
         else:
             state = lax.while_loop(greedy_search_cond_fn, greedy_search_body_fn, state)
 
-        return FlaxGreedySearchOutput(sequences=state.sequences)
+        if output_scores:
+            final_scores = state.scores
+        else:
+            final_scores = None
+
+        return FlaxGreedySearchOutput(sequences=state.sequences, scores=final_scores)
```

Comment:
Note that `state.scores` will be `None` if `output_scores=False`, so we can just pass this value directly
```suggestion
        return FlaxGreedySearchOutput(sequences=state.sequences, scores=state.scores)
```